projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/deploy/DataJobsSynchronizer.java

```diff
@@ -144,6 +144,8 @@ void synchronizeDataJob(
           desiredDataJobDeployment,
           actualDataJobDeployment,
           isDeploymentPresentInKubernetes);
+    } else if (actualDataJobDeployment != null) {
```

Comment:
Where are the tests that verify `synchronizeDataJobs` (all jobs) work as expected? I see tests that `synchronizeDataJob` (singular) works. But not for the full sync.

I am suspicious deleting job might not work because we are iterating using `dataJobsFromDB` but if we delete a data job then `dataJobsFromDB` would not contain our deleted job.

Reply:
Correct, the synchronizeDataJobs does not cover this case.

Reply:
Are you ok to implement this in a separate PR?

Reply:
Yeah.